src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -52,24 +55,43 @@ protected override void OnBalanceChanging(ApplicationEngine engine, UInt160 acco
             StorageKey key = CreateStorageKey(Prefix_Candidate).Add(state.VoteTo);
             CandidateState candidate = engine.Snapshot.Storages.GetAndChange(key).GetInteroperable<CandidateState>();
             candidate.Votes += amount;
-            CheckCandidate(engine.Snapshot, key, candidate);
+            CheckCandidate(engine.Snapshot, state.VoteTo, candidate);
         }
 
         private void DistributeGas(ApplicationEngine engine, UInt160 account, NeoAccountState state)
         {
             // PersistingBlock is null when running under the debugger
             if (engine.Snapshot.PersistingBlock == null) return;
 
-            BigInteger gas = CalculateBonus(engine.Snapshot, state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
+            BigInteger gas = CalculateBonus(engine.Snapshot, state.VoteTo, state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
             state.BalanceHeight = engine.Snapshot.PersistingBlock.Index;
             GAS.Mint(engine, account, gas);
         }
 
-        private BigInteger CalculateBonus(StoreView snapshot, BigInteger value, uint start, uint end)
+        private BigInteger CalculateBonus(StoreView snapshot, ECPoint vote, BigInteger value, uint start, uint end)
         {
             if (value.IsZero || start >= end) return BigInteger.Zero;
             if (value.Sign < 0) throw new ArgumentOutOfRangeException(nameof(value));
 
+            BigInteger neoHolderReward = CalculateNeoHolderReward(snapshot, value, start, end);
+            if (vote is null) return neoHolderReward;
+
+            StorageKey keyStart = CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(vote).AddBigEndian(start);
+            var (key, item) = snapshot.Storages.Seek(keyStart.ToArray(), SeekDirection.Backward).FirstOrDefault();
```

Comment:
Use `FindRange()`?

Reply:
No, `FindRange` use `Forward` order.

Reply:
We can add a parameter to `FindRange` to specify the direction.

Reply:
In fact `FindRange` doesn't sort the items

```csharp
            myDataCache.Add(new MyKey("key1"), new MyValue("value1"));
            myDataCache.Add(new MyKey("key2"), new MyValue("value2"));

            myDataCache.InnerDict.Add(new MyKey("key4"), new MyValue("value4")); // if we cache first the key4 it will fail
            myDataCache.InnerDict.Add(new MyKey("key3"), new MyValue("value3"));

            var items = myDataCache.FindRange(new MyKey("key3"), new MyKey("key5")).ToArray();
            items[0].Key.Should().Be(new MyKey("key3"));
            items[0].Value.Should().Be(new MyValue("value3"));
            items[1].Key.Should().Be(new MyKey("key4"));
            items[1].Value.Should().Be(new MyValue("value4"));
            items.Count().Should().Be(2);
```

Reply:
I'll fix it.

Reply:
See https://github.com/neo-project/neo/pull/1954